cmip6_downscaling/tasks/pyramid.py

```diff
@@ -0,0 +1,92 @@
+import dask
+import datatree as dt
+import fsspec
+import xarray as xr
+from carbonplan_data.metadata import get_cf_global_attrs
+from carbonplan_data.utils import set_zarr_encoding
+from ndpyramid import pyramid_regrid
+from prefect import task
+
+PIXELS_PER_TILE = 128
+
+
+def _load_coords(ds: xr.Dataset) -> xr.Dataset:
+    '''Helper function to explicitly load all dataset coordinates'''
+    for var, da in ds.coords.items():
+        ds[var] = da.load()
+    return ds
+
+
+def postprocess(dt: dt.DataTree, levels: int) -> dt.DataTree:
+    '''Postprocess data pyramid
+
+    Adds multiscales metadata and sets Zarr encoding
+
+    Parameters
+    ----------
+    dt : dt.DataTree
+        Input data pyramid
+    levels : int
+        Number of levels in pyramid
+
+    Returns
+    -------
+    dt.DataTree
+        Updated data pyramid with metadata / encoding set
+    '''
+
+    for level in range(levels):
+        slevel = str(level)
+        dt.ds.attrs['multiscales'][0]['datasets'][level]['pixels_per_tile'] = PIXELS_PER_TILE
+
+        dt[slevel].ds = dt[slevel].ds.chunk(
+            {"x": PIXELS_PER_TILE, "y": PIXELS_PER_TILE, "time": 31}
+        )
+        dt[slevel].ds['date_str'] = dt[slevel].ds['date_str'].chunk(-1)
+
+        dt[slevel].ds = set_zarr_encoding(
+            dt[slevel].ds, codec_config={"id": "zlib", "level": 1}, float_dtype="float32"
+        )
+        dt[slevel].ds['time'].encoding['dtype'] = 'int32'
+        dt[slevel].ds['time_bnds'].encoding['dtype'] = 'int32'
+    dt.ds.attrs.update(**get_cf_global_attrs())
+    return dt
+
+
+@task(log_stdout=True)
+# Question: How should we
+# tags=['dask-resource:TASKSLOTS=1']
```

Comment:
It seems like this tasks called `pyramid_regrid`, which uses xesmf. Do you want to keep the tags['dask-resource:TASKSLOTS=1'] in?